api/renewer.go

```diff
@@ -183,31 +187,41 @@ func (r *Renewer) renewAuth() error {
 		default:
 		}
 
-		// Renew the auth.
-		renewal, err := client.Auth().Token().RenewTokenAsSelf(token, r.increment)
-		if err != nil {
-			return err
-		}
+		var leaseDuration time.Duration
+
+		switch {
+		case nonRenewable:
+			// Can't renew, just keep same expiration so we exit when it's
+			// reauthentication time
+			leaseDuration = initialTime.Add(priorDuration).Sub(time.Now())
 
 		// Push a message that a renewal took place.
-		select {
-		case r.renewCh <- &RenewOutput{time.Now().UTC(), renewal}:
 		default:
+			renewal, err := client.Auth().Token().RenewTokenAsSelf(token, r.increment)
+			if err != nil {
+				return err
+			}
+
+			// Push a message that a renewal took place.
+			select {
+			case r.renewCh <- &RenewOutput{time.Now().UTC(), renewal}:
+			default:
+			}
+
+			// Somehow, sometimes, this happens.
+			if renewal == nil || renewal.Auth == nil {
+				return ErrRenewerNoSecretData
+			}
+
+			// Do nothing if we are not renewable
+			if !renewal.Auth.Renewable {
```

Comment:
Does this still make sense? Seems odd that it's okay to start with a non-renewable token, but if you end up with one in the course of renewing, you'll get an error?

Reply:
It shouldn't ever change so for now this should be fine.